### PR TITLE
Clean up plots and add init conditions to API docs.

### PR DESCRIPTION
**Pull request by akshaysridhar**
Use Makie in the plot pipeline. (Updates to 3D box config in future PR.) 
Add API docs for initial conditions and post-proc functions.
Fix error in docstring + add bib entry. 

Partially addresses #1399.

### CODE REVIEW

File: post_processing/plot/plot_helpers.jl [hunk 1]
```diff
@@ -445,13 +410,17 @@ function generate_paperplots_held_suarez(fig_dir, nc_files; moist)
             global lon = nc["lon"][:]
             global z = nc["z"][:]
             global u = nc["u"][:]
+            global v = nc["v"][:]
```
Comment (charleskawczynski) [inline post_processing/plot/plot_helpers.jl:413]: 👀
Why are we using `global`s here?

Comment (akshaysridhar) [reply post_processing/plot/plot_helpers.jl:413]: I think this is being done for the time mean plots? I was wondering the same myself; the intent of this PR is simply to replace `Makie` as our primary plot package.

Comment (szy21) [reply post_processing/plot/plot_helpers.jl:410]: I think we just need `global` to be able to access these variables in the `else` block? I'm not 100% sure.

Comment (akshaysridhar) [reply post_processing/plot/plot_helpers.jl:410]: Are `begin` blocks + `push!` better here than `global` + `cat` ?

Comment (charleskawczynski) [reply post_processing/plot/plot_helpers.jl:411]: It's fine to clean up later, I was just surprised to see this

Comment (akshaysridhar) [reply post_processing/plot/plot_helpers.jl:411]: Noted here: https://github.com/CliMA/ClimaAtmos.jl/issues/1816

Comment (charleskawczynski) [reply post_processing/plot/plot_helpers.jl:411]: Thanks!

**Review by bischtob — APPROVED** [review]
LGTM!

File: post_processing/plot/plot_helpers.jl [hunk 2]
```diff
@@ -22,42 +81,31 @@ function generate_paperplots_dry_baro_wave(fig_dir, nc_files)
...
-                fig_dir * "/dbw_day$day.png",
+            create_plot!(
+                fig;
+                p_loc = (3, 1),
+                X = lon[lonidx],
+                Y = lat[latidx],
+                Z = vort[lonidx, latidx, 1, 1] .* FT(1e5),
+                title = "Vorticity(× 10⁵): Day $day; z $(round(z[1])) m",
             )
+            CairoMakie.save(fig_dir * "/mbw_day$day.png", fig)
```
Comment (szy21) [inline post_processing/plot/plot_helpers.jl:108]: `dbw_day$day.png` @akshaysridhar:)

Comment (akshaysridhar) [reply post_processing/plot/plot_helpers.jl:108]: @szy21 https://github.com/CliMA/ClimaAtmos.jl/pull/1825

**Pull request merged**